src/lambda/commands/deploySamApplication.ts

```diff
@@ -40,7 +43,17 @@ export async function deploySamApplication({
         const buildTemplatePath = path.join(buildDestination, 'template.yaml')
         const outputTemplatePath = path.join(tempFolder, 'template.yaml')
         let stage: 'starting up' | 'building' | 'packaging' | 'deploying' = 'starting up'
+
+        const settingsConfiguration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+        const profile = settingsConfiguration.readSetting<string>(profileSettingKey)
+        let msg = ''
+
         try {
+            if (!profile) {
+                const err = new Error('No AWS profile selected')
```

Comment:
It might be worth making a custom error like `NoActiveCredentialError` and throwing that. This would allow us to react with specific messaging/handling.

Reply:
✔️ , open to changing the location of the error (added to `awsContext.ts` because it should be called in conjunction with `getCredentialProfileName`)